health/records.go

```diff
@@ -0,0 +1,76 @@
+package health
+
+import (
+	"container/list"
+	"os"
+)
+
+type records struct {
+	capacity int
+	items    *list.List
+}
+
+func newRecords(capacity int) *records {
+	return &records{
+		capacity: capacity,
+		items:    list.New(),
+	}
+}
+
+func (records *records) addRecord(incomingRecord record) {
+	if !records.makeRoomForRecord(incomingRecord) {
+		return
+	}
+
+	insertionPlace := records.findInsertionPlace(incomingRecord)
```

Comment:
is this working properly with concurrent addRecord? I think you would need a mutex locked before finding the place of insertion and unlocked after inserting.

edit: Maybe a mutex lock before making room for record

Reply:
Added mutex.